libr/syscall/sysregs.c

```diff
@@ -0,0 +1,74 @@
+/* radare - Copyright 2016 - LGPL -- xvilka */
+
+#include <r_types.h>
+#include <r_util.h>
+#include <r_db.h>
+#include <r_syscall.h>
+#include <stdio.h>
+#include <string.h>
+
+R_API RSysregs* r_sysregs_new() {
+	RSysregs *rs = R_NEW0 (RSysregs);
+	if (rs) {
+		rs->arch = "x86";
+		rs->cpu = "x86";
+		rs->sysregs = NULL;
+	}
+	return rs;
+}
+
+R_API void r_sysregs_free(RSysregs *s) {
+	sdb_free (s->db);
+	free (s->arch);
+	free (s->cpu);
+	free (s->sysregs);
+	memset (s, 0, sizeof (RSysregs));
+	free (s);
+}
+
+R_API int r_sysregs_setup(RSysregs *s, const char *arch, const char *cpu) {
+	const char *file;
+	if (!arch) arch = R_SYS_ARCH;
+
+	s->cpu = strdup (cpu);
+
+	if (!strcmp (cpu, "any")) { // ignored
+		return true;
+	}
+
+#define SYSREGSPATH R2_LIBDIR"/radare2/"R2_VERSION"/sysregs"
```

Comment:
local defines at the beginning of the function frame, like local vars or just use const char * for this